src/components/molecules/Chatbox/Chatbox.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useEffect, useState } from "react";
+import firebase from "firebase/app";
+import ChatboxMessageList from "./ChatboxMessageList";
+import { useForm } from "react-hook-form";
+import "./CampChat.scss";
+import { useVenueId } from "hooks/useVenueId";
+import "./Chatbox.scss";
+import {
+  PrivateChatMessage,
+  RestrictedChatMessage,
+} from "components/context/ChatContext";
+import { WithId } from "utils/id";
+
+interface ChatOutDataType {
+  messageToTheBand: string;
+}
+
+interface PropsType {
+  chats: WithId<PrivateChatMessage | RestrictedChatMessage>[];
+  onMessageSubmit: (data: ChatOutDataType) => void;
+  allowDelete?: boolean;
+  emptyListMessage?: string;
+}
+
+const ChatBox: React.FC<PropsType> = ({
+  allowDelete,
+  chats,
+  onMessageSubmit,
+  emptyListMessage,
+}) => {
+  const venueId = useVenueId();
+  const [isMessageToTheBarSent, setIsMessageToTheBarSent] = useState(false);
+
+  useEffect(() => {
+    if (isMessageToTheBarSent) {
+      setTimeout(() => {
+        setIsMessageToTheBarSent(false);
+      }, 2000);
+    }
+  }, [isMessageToTheBarSent]);
+
+  const { register, handleSubmit, reset } = useForm<ChatOutDataType>({
+    mode: "onSubmit",
+  });
+
+  const onBarMessageSubmit = async (data: ChatOutDataType) => {
+    setIsMessageToTheBarSent(true);
+    onMessageSubmit(data);
+    reset();
+  };
+
+  const deleteMessage = async (id: string) => {
+    await firebase
+      .firestore()
+      .doc(`venues/${venueId}/chats/${id}`)
+      .update({ deleted: true });
+  };
+
+  return (
+    <div className="chat-container show">
+      {chats && (
+        <ChatboxMessageList
+          messages={chats}
+          emptyListMessage={emptyListMessage}
+          allowDelete={allowDelete}
+          deleteMessage={deleteMessage}
+        />
+      )}
+      <form className="chat-form" onSubmit={handleSubmit(onBarMessageSubmit)}>
```

Comment:
Still unsure if this is an issue or not.

Reply:
Looks like the library is already using useCallback for the most of the props inc. handleSubmit.
https://github.com/react-hook-form/react-hook-form/blob/master/src/useForm.ts#L1305

Reply:
Yep I think this is OK to not refactor, otherwise we are basically saying we need to refactor every function used anonymously as a prop, which is a lot..

Reply:
> @thecadams otherwise we are basically saying we need to refactor every function used anonymously as a prop, which is a lot..

Yet in most places where we are doing it, that's exactly what we'll need to do to fix the re-renders properly.